wagtail_newsletter/models.py

```diff
@@ -12,7 +17,98 @@ class Meta:  # type: ignore
     def __str__(self):
         return self.name
 
+    @property
+    def member_count(self):
+        if self.segment:
+            try:
+                return audiences.AudienceSegment.objects.get(
+                    pk=self.segment
+                ).member_count
+            except audiences.AudienceSegment.DoesNotExist:
+                return None
+
+        elif self.audience:
+            try:
+                return audiences.Audience.objects.get(pk=self.audience).member_count
+            except audiences.Audience.DoesNotExist:
+                return None
+
+        else:
+            return None
+
 
 class NewsletterRecipients(NewsletterRecipientsBase):
     class Meta:  # type: ignore
         verbose_name_plural = "Newsletter recipients"
+
+
+class NewsletterPageMixin(Page):
+    base_form_class: type
+
+    newsletter_recipients = models.ForeignKey(
+        get_recipients_model_string(),
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    newsletter_subject = models.CharField(
+        max_length=1000,
+        blank=True,
+        help_text="Subject for the newsletter. Defaults to page title if blank.",
+    )
+    newsletter_campaign = models.CharField(max_length=1000, blank=True)
+
+    class Meta:  # type: ignore
+        abstract = True
+
+    @classmethod
+    def get_newsletter_panels(cls):
```

Comment:
Unlike Wagtail's built-in tabs (`content_panels`, `promote_panels` etc), this one is implemented as a getter method, because it requires the `recipients_chooser_viewset` object from the `viewsets` model. Writing it declaratively would result in a circular model import.

It could have been a class property, I guess, but that might have been more confusing, and difficult to override by a subclass.